modified code for file permission access

sgupta3112 opened the pull request:
Hi @rcvrgs ,

I have again raise a pull request and branched out with dev as well. 

Please check it and let me know if any changes are required. 

Shivam Gupta

sgupta3112 committed:
modified code for file permission access

brol1dev commented on FreewheelSampleApp/app/src/main/java/com/ooyala/sample/players/CustomConfiguredFreewheelPlayerActivity.java:
```diff
@@ -52,6 +58,36 @@ public final static String getName() {
   @Override
   public void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
+    // Here, thisActivity is the current activity
```
don't get this comment, I think you're talking about the `this` keyword, you can remove the comment if that's the case

sgupta3112 replied:
Done. Removed the comment.

brol1dev commented on FreewheelSampleApp/app/src/main/java/com/ooyala/sample/players/PreconfiguredFreewheelPlayerActivity.java:
```diff
@@ -53,6 +60,36 @@ public final static String getName() {
+            != PackageManager.PERMISSION_GRANTED) {
+
+      // Should we show an explanation?
+      if (ActivityCompat.shouldShowRequestPermissionRationale(this,
```
This is a sample app, so we don't need this method. What this does is let us know that the user denied the permission previously and we could decide to show a different UI to explain them what we need it for.

sgupta3112 replied:
ok. Can you please fix it as per the standard. Feel free to modify it. our main objective to give permission to the sample app for write log in logFile which get created in external storage while running the automation.

brol1dev commented on FreewheelSampleApp/app/src/main/java/com/ooyala/sample/players/PreconfiguredFreewheelPlayerActivity.java:
```diff
@@ -53,6 +60,36 @@ public final static String getName() {
+        // this thread waiting for the user's response! After the user
+        // sees the explanation, try again to request the permission.
+
+        ActivityCompat.requestPermissions(this,
```
When you remove `shouldShowRequestPermissionRationale` you should only have one `requestPermissions`

sgupta3112 replied:
Yes, We have done it.

brol1dev commented on FreewheelSampleApp/app/src/main/java/com/ooyala/sample/players/PreconfiguredFreewheelPlayerActivity.java:
```diff
@@ -24,14 +28,17 @@
+ *
  */
 public class PreconfiguredFreewheelPlayerActivity extends Activity implements Observer {
+  private static final int MY_PERMISSIONS_REQUEST_READ_CONTACTS = 0;
```
Rename this to something correct and meaningful. We want permissions to `WRITE_EXTERNAL_STORAGE` and you're naming this variable as `REQUEST_READ_CONTACTS` which is intent for a different permission when I read it. This is for developer friendliness, so name it to something like `PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE`

sgupta3112 replied:
Name has been changed.

brol1dev commented on FreewheelSampleApp/app/src/main/java/com/ooyala/sample/players/PreconfiguredFreewheelPlayerActivity.java:
```diff
@@ -24,14 +28,17 @@
  */
 public class PreconfiguredFreewheelPlayerActivity extends Activity implements Observer {
+  private static final int MY_PERMISSIONS_REQUEST_READ_CONTACTS = 0;
+  private String text;
```
instead of adding this text variable here, do something else. For example, use a boolean flag that will be `true` when you get the permission to `WRITE_EXTERNAL_STORAGE`

sgupta3112 replied:
In Text we are taking Ooyala Sample App's event and writing in logFile.

brol1dev commented on FreewheelSampleApp/app/src/main/java/com/ooyala/sample/players/PreconfiguredFreewheelPlayerActivity.java:
```diff
@@ -109,11 +146,32 @@ public void update(Observable arg0, Object argN) {
+
+          // permission was granted, yay! Do the
+          // contacts-related task you need to do.
+          Playbacklog.writeToSdcardLog(text);
```
This is out of context, if you use a boolean flag as I told you before, you would be setting it to `true` here. Then whenever you want to use `Playbacklog.writeToSdcardLog(aString);` you should first verify if you have the permission to write to the SdCard.

brol1dev reviewed (dismissed):
Added some changes inline, but I'm going to ask you for another one here.

Since you are just copying and pasting the same logic in different Activities, please create a parent Activity for all of the PlayerActivities that will implement the same logic for all of them.

sgupta3112 commented:
We have done few changes and pushed the code. Now please review it again and please also feel free to do modification if you think.

shivam3112 committed:
Modified or update the code as per the developers suggestions

Commit:
Add AbstractHookActivity

The player activities extend AbstractHookActivity. AbstractHookActivity is used ask for WRITE_EXTERNAL_STORAGE permission. Refactor the player activities to have less code.

Commit:
Update OptionsFreewheelPlayerActivity

brol1dev reviewed (commented):
I need someone else to review changes, since I updated this code.

shivam3112 committed:
added setUseExoPlayer code

akshaydch11 commented:
Looking good to me 👍
